src/hooks/useApi.ts

```diff
@@ -0,0 +1,40 @@
+import { useEffect, useState } from 'react';
+
+const BASE_URL = '/api';
+
+/**
+ * Fetches resources from an API endpoint and returns them.
+ *
+ * @param path the path to the endpoint to fetch
+ * @returns the error and the result returned from the endpoint.
+ *    Only one of these will be set at any time.
+ */
+const useApi = <T>(path: string): [number | null, T | null] => {
+  const [result, setResult] = useState<T | null>(null);
+  const [error, setError] = useState<number | null>(null);
+
+  useEffect(() => {
+    const sendRequest = async () => {
+      setResult(null);
+      setError(null);
+
+      try {
+        const response = await fetch(`${BASE_URL}${path}`);
```

Comment:
Diese `async/await` Methoden sind eigentlich nur eine andere Schreibweise für Promises. Falls ihr beides nicht kennt: Da geht es vor allem darum, mit asynchronen Prozessen zu arbeiten (also zum Beispiel eben Request/Response). Auch wenn es hier linear aussieht, wartet `await` quasi so lange, bis die `response` auch wirklich da ist. Das ist aber non-blocking, das heißt, dass alle anderen Render-Prozesse schon weiterlaufen. In diesem Fall hat das zur Folge, das erstmal eine leer Liste gerendert wird. Erst wenn die Response dann da ist, wird erneut gerendert, und die Liste wird befüllt.